app/scripts/modules/core/src/pipeline/config/stages/jenkins/jenkinsStage.html

```diff
@@ -65,9 +65,38 @@ <h4 class="text-left" >Job Parameters</h4>
         <b class="break-word">{{parameter.name}}</b>
         <help-field content="{{parameter.description}}" ng-if="parameter.description"></help-field>
       </div>
-      <div class="col-md-5">
-        <input disabled ng-if="useDefaultParameters[parameter.name]" type="text" class="form-control input-sm" value="{{parameter.defaultValue}}"/>
-        <input ng-if="!useDefaultParameters[parameter.name]" type="text" class="form-control input-sm" ng-model="userSuppliedParameters[parameter.name]" ng-change="updateParam(parameter.name)"/>
+      <div class="col-md-5" ng-switch on="parameter.type">
+        <div ng-switch-when="PasswordParameterDefinition">
+          <!-- Passwords do not expose their defaults, for obvious reasons, so we can skip this -->
+          <input ng-if="!useDefaultParameters[parameter.name]" type="password" class="form-control input-sm" ng-model="userSuppliedParameters[parameter.name]" ng-change="updateParam(parameter.name)"/>
+        </div>
+        <div ng-switch-when="BooleanParameterDefinition">
+          <input disabled ng-if="useDefaultParameters[parameter.name] && parameter.defaultValue == 'true'" type="checkbox" class="input-sm" checked/>
+          <input disabled ng-if="useDefaultParameters[parameter.name] && parameter.defaultValue == 'false'" type="checkbox" class="input-sm"/>
```

Comment:
It doesn't appear to like it. I seem to recall trying this when I was first setting the boolean section up.

![image](https://user-images.githubusercontent.com/3680008/28837877-5d931fce-76bc-11e7-8d2f-bdaa4dc09b47.png)